MediaBrowser.MediaEncoding/Probing/ProbeResultNormalizer.cs

```diff
@@ -746,6 +741,19 @@ private MediaStream GetMediaStream(bool isAudio, MediaStreamInfo streamInfo, Med
                 stream.AverageFrameRate = GetFrameRate(streamInfo.AverageFrameRate);
                 stream.RealFrameRate = GetFrameRate(streamInfo.RFrameRate);
 
+                // Some interlaced H.264 files in mp4 containers using MBAFF coding aren't flagged as being interlaced by FFprobe,
+                // so for H.264 files we also check if the codec timebase duration is half the reported frame rate duration to
+                // determine if the file is interlaced
+                bool videoInterlaced = !string.IsNullOrWhiteSpace(streamInfo.FieldOrder)
+                    && !string.Equals(streamInfo.FieldOrder, "progressive", StringComparison.OrdinalIgnoreCase);
+                bool h264MbaffCoded = string.Equals(stream.Codec, "h264", StringComparison.OrdinalIgnoreCase)
+                    && string.IsNullOrWhiteSpace(streamInfo.FieldOrder)
+                    && 1f / (stream.AverageFrameRate * 2) == GetFrameRate(stream.CodecTimeBase);
```

Comment:
Using floating point equality here will [probably not be accurate](https://bitbashing.io/comparing-floats.html) all of the time.

Is there another way to do this calculation?

Reply:
Unfortunately there isn't really any other way to do the calculation, as FFprobe returns fractional values.

However I have changed it to round the values, which should hopefully mitigate this issue.